### PR TITLE
Zone json export

### PR DESCRIPTION
**Pull request by CBiasuzzi**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**
adds a json export to the zone graph;  adds more substation graph's data to its json export;  extract nodes structures from the cells, so that, at the same level, we have nodes, cells, edges

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
Sorry christian I forgot to review this PR. Could you merge it with master?

**Comment by CBiasuzzi** [comment]
So sorry I totally missed this:  I've merged the branch with master.

File: single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java [hunk 1]
```diff
@@ -50,16 +50,28 @@ public void test() {
         // write Json and compare to reference (with smart substation layout)
         SubstationGraph g = graphBuilder.buildSubstationGraph(substation.getId(), false);
         new ForceSubstationLayoutFactory(ForceSubstationLayoutFactory.CompactionType.NONE).create(g, new PositionVoltageLevelLayoutFactory()).run(layoutParameters);
-        assertEquals(toString("/TestCase11SubstationGraphSmart.json"), toJson(g, "/TestCase11SubstationGraphSmart.json", false));
+        //assertEquals(toString("/TestCase11SubstationGraphSmart.json"), toJson(g, "/TestCase11SubstationGraphSmart.json", false));
```
Comment (FranckLecuyer) [inline single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java:53]: why is this line commented now ?

File: single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java [hunk 2]
```diff
@@ -50,16 +50,28 @@ public void test() {
...
+        assertEquals(substation.getId(), g.getSubstationId());
+        assertEquals(substation.getVoltageLevelStream().count(), g.getNodes().size());
+        assertEquals(19, g.getEdges().size());
+        assertEquals(8, g.getMultiTermNodes().size());
 
         // write Json and compare to reference (with smart substation layout and horizontal compaction)
         g = graphBuilder.buildSubstationGraph(substation.getId(), false);
         new ForceSubstationLayoutFactory(ForceSubstationLayoutFactory.CompactionType.HORIZONTAL).create(g, new PositionVoltageLevelLayoutFactory()).run(layoutParameters);
-        assertEquals(toString("/TestCase11SubstationGraphSmartHorizontal.json"), toJson(g, "/TestCase11SubstationGraphSmartHorizontal.json", false));
+        //assertEquals(toString("/TestCase11SubstationGraphSmartHorizontal.json"), toJson(g, "/TestCase11SubstationGraphSmartHorizontal.json", false));
```
Comment (FranckLecuyer) [inline single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java:62]: why is this line commented now ?

File: single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java [hunk 3]
```diff
@@ -50,16 +50,28 @@ public void test() {
...
+        assertEquals(substation.getId(), g.getSubstationId());
+        assertEquals(substation.getVoltageLevelStream().count(), g.getNodes().size());
+        assertEquals(19, g.getEdges().size());
+        assertEquals(8, g.getMultiTermNodes().size());
 
         // write Json and compare to reference (with smart substation layout and vertical compaction)
         g = graphBuilder.buildSubstationGraph(substation.getId(), false);
         new ForceSubstationLayoutFactory(ForceSubstationLayoutFactory.CompactionType.VERTICAL).create(g, new PositionVoltageLevelLayoutFactory()).run(layoutParameters);
-        assertEquals(toString("/TestCase11SubstationGraphSmartVertical.json"), toJson(g, "/TestCase11SubstationGraphSmartVertical.json", false));
+        //assertEquals(toString("/TestCase11SubstationGraphSmartVertical.json"), toJson(g, "/TestCase11SubstationGraphSmartVertical.json", false));
```
Comment (FranckLecuyer) [inline single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java:71]: why is this line commented now ?

File: single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java (hunk 1 above)
Comment (CBiasuzzi) [reply single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java]: forgive me:  comment removed.

File: single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java (hunk 2 above)
Comment (CBiasuzzi) [reply single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java]: forgive me:  comment removed.

File: single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java (hunk 3 above)
Comment (CBiasuzzi) [reply single-line-diagram-force-layout/src/test/java/com/powsybl/sld/force/layout/TestCaseSmartSubstationGraph.java]: forgive me:  comment removed.

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**